version: '3.3'

services:
  client:
    image: qbonaventure/discord-bot-ws:latest
    secrets:
      - source: ftcbot-broker-password
        target: /run/secrets/ftcbot-broker-password
        mode: 0777
      - ftcbot-broker-username
      - source: ftcbot-discord-auth-token
        target: /run/secrets/ftcbot-discord-auth-token
    ports:
      - "7632:7632"
    tty: true
    restart: always
    env_file:
      - .env
    links: 
      - broker
      - websocket-cache
    deploy:
      labels :
        ftcbot.service: webservice
      replicas: 1
      restart_policy:
        condition: on-failure
    networks:
      ftcbot:
    
  broker:
    image: rabbitmq:3.7-alpine
    expose:
      - "4369"
      - "5671"
      - "5672"
      - "25672"
    deploy:
      labels:
        ftcbot.service: broker
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      restart_policy:
        condition: on-failure
    networks:
      ftcbot:
      
  websocket-cache:
    image: redis:alpine
    expose:
      - "6379"
    volumes:
      -  websocket_cache_data:/data
    deploy:
      labels:
        ftcbot.service: websocket-cache
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      restart_policy:
        condition: on-failure
    networks:
      ftcbot:
      
secrets:
  ftcbot-broker-password:
    external: true
  ftcbot-broker-username:
    external: true
  ftcbot-discord-auth-token:
    external: true
      
networks:
  ftcbot:
 
volumes:
  websocket_cache_data:
 